test/extended/builds/s2i_extended_build.go

```diff
@@ -0,0 +1,140 @@
+
+			g.By("waiting for the build to complete")
+			err = exutil.WaitForABuild(oc.REST().Builds(oc.Namespace()), buildName, exutil.CheckBuildSuccessFn, exutil.CheckBuildFailedFn)
+			if err != nil {
+				exutil.DumpBuildLogs(buildConfigName, oc)
+			}
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			buildLog, err := oc.Run("logs").Args("-f", "build/"+buildName).Output()
+			if err != nil {
+				e2e.Failf("Failed to fetch build logs of %q: %v", buildLog, err)
+			}
+
+			g.By("expecting that .s2i/bin/assemble-runtime was executed")
+			o.Expect(buildLog).To(o.ContainSubstring(`Using "assemble-runtime" installed from "https://raw.githubusercontent.com/php-coder/java-maven-hello-world/s2i/.s2i/bin/assemble-runtime"`))
+			o.Expect(buildLog).To(o.ContainSubstring(".s2i/bin/assemble-runtime: assembling app within runtime image"))
+
+			g.By("expecting that environment variable from BuildConfig is available")
+			o.Expect(buildLog).To(o.ContainSubstring(".s2i/bin/assemble-runtime: USING_ENV_FROM_BUILD_CONFIG=yes"))
+
+			g.By("expecting that environment variable from .s2i/environment isn't available")
+			o.Expect(buildLog).To(o.ContainSubstring(".s2i/bin/assemble-runtime: USING_ENV_FROM_FILE=no"))
+		})
+	})
+
+	g.Describe("with scripts from runtime image", func() {
+		oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+		g.It("should use assemble-runtime script from that image", func (){
+			const buildConfigName = "java-extended-build-from-image"
+			const buildName = buildConfigName + "-1"
+
+			g.By("creating build config")
+			err := oc.Run("create").Args("-f", scriptsFromImageBc).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("running the build")
+			err = oc.Run("start-build").Args(buildConfigName).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("waiting for the build to complete")
+			err = exutil.WaitForABuild(oc.REST().Builds(oc.Namespace()), buildName, exutil.CheckBuildSuccessFn, exutil.CheckBuildFailedFn)
+			if err != nil {
+				exutil.DumpBuildLogs(buildConfigName, oc)
+			}
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			buildLog, err := oc.Run("logs").Args("-f", "build/"+buildName).Output()
+			if err != nil {
+				e2e.Failf("Failed to fetch build logs of %q: %v", buildLog, err)
+			}
+
+			g.By("expecting that .s2i/bin/assemble-runtime was executed")
+			o.Expect(buildLog).To(o.ContainSubstring(`Using "assemble-runtime" installed from "image:///usr/libexec/s2i/assemble-runtime"`))
+			o.Expect(buildLog).To(o.ContainSubstring(".s2i/bin/assemble-runtime: assembling app within runtime image"))
+
+			g.By("expecting that environment variable from BuildConfig is available")
+			o.Expect(buildLog).To(o.ContainSubstring(".s2i/bin/assemble-runtime: USING_ENV_FROM_BUILD_CONFIG=yes"))
+
+			g.By("expecting that environment variable from .s2i/environment isn't available")
+			o.Expect(buildLog).To(o.ContainSubstring(".s2i/bin/assemble-runtime: USING_ENV_FROM_FILE=no"))
+		})
+	})
+
+})
```

Comment:
@bparees @PI-Victor Ideally I would like to have a tests for testing `run` script also. Right now they're focusing on testing `assemble-runtime` script only.